### PR TITLE
Fix date format for `InputAge` component

### PR DESCRIPTION
**Pull request by siddh1004**

### CODE REVIEW

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputAge.kt [hunk 1]
```diff
@@ -53,6 +59,8 @@ import java.util.Calendar
  * @param uiModel: data class [InputAgeModel] with all parameters for component.
  * @param modifier: optional modifier.
  */
+@Suppress("DEPRECATION")
+@Deprecated("This component is deprecated and will be removed in the next release. Use InputDateTime instead.")
```
Comment (xavimolloy) [inline designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputAge.kt:63]: Could you modify the reason for the deprecation?

File: designsystem/src/androidUnitTest/kotlin/org/hisp/dhis/mobile/ui/designsystem/InputAgeSnapshotTest.kt [hunk 2]
```diff
@@ -22,74 +23,81 @@ class InputAgeSnapshotTest {
...
                 InputAge(
-                    InputAgeModel(
-                        title = "Label",
+                    state = rememberInputAgeState(
+                        inputAgeData = InputAgeData(
+                            title = "Label",
+                        ),
                         inputType = AgeInputType.DateOfBirth(
                             TextFieldValue("01011985"),
                         ),
```
Comment (xavimolloy) [inline designsystem/src/androidUnitTest/kotlin/org/hisp/dhis/mobile/ui/designsystem/InputAgeSnapshotTest.kt:55]: Could you add some examples of the input receiving the expected value? the new implementation should receive a value like  "1991-11-27" format it internally to "27111991" and return the first format again whenever the value changes

**Review by xavimolloy — CHANGES_REQUESTED** [review]

**Review by xavimolloy — CHANGES_REQUESTED** [review]
Also, do you think you could create or adapt some existing unit tests to verify that the component is formating date values correctly and will not crash if the value is in an unexpected format?

File: designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputAge.kt (hunk 1 above)
Comment (siddh1004) [reply designsystem/src/commonMain/kotlin/org/hisp/dhis/mobile/ui/designsystem/component/InputAge.kt:63]: Done.

**Review by xavimolloy — APPROVED** [review]

**Review by andresmr — APPROVED** [review]

**Pull request merged**